core/kernel/user_ta.c

```diff
@@ -159,9 +172,12 @@ static TEE_Result user_ta_enter(struct ts_session *session,
 	usr_stack -= ROUNDUP(sizeof(struct utee_params), STACK_ALIGNMENT);
 	usr_params = (struct utee_params *)usr_stack;
 	if (ta_sess->param)
-		init_utee_param(usr_params, ta_sess->param, param_va);
+		res = init_utee_param(usr_params, ta_sess->param, param_va);
 	else
-		memset(usr_params, 0, sizeof(*usr_params));
+		res = clear_user(usr_params, sizeof(*usr_params));
+
+	if (res)
+		goto out_pop_session;
```

Comment:
`vm_clean_param()` must be called `if (ta_sess->param)`, but not `update_from_utee_param()` when exiting from here.

Reply:
updated. may look ugly though 🥲